app/src/main/java/com/example/drawingactivity/CustomColorViewModel.kt

```diff
@@ -36,11 +38,28 @@ class CustomColorViewModel : ViewModel() {
         currentBitmap = b
     }
 
+    var isDialogOpen = false
+    var currentUserEnteredLayerName: String? = null
+    var layerNumber: Int = 1
+
+    fun addLayer(s: String, b: Boolean) {
+        if (s.length < NAME_LENGTH) {
+            if (b) {
+                layerNumber += 1
```

Comment:
Is this going to work if layers are deleted? Or renamed? Or if the file is saved and reloaded?

Reply:
Implemented more robust logic for adding layers based on what names are in use and which are not.